pmd-core/src/test/java/net/sourceforge/pmd/RuleSetFactoryTest.java

```diff
@@ -349,6 +349,42 @@ public void testRuleSetReferencesRulesetWithAExcludedNonExistingRule() throws Ex
                     "WARNING: Unable to exclude rules [NonExistingRule] from ruleset reference rulesets/dummy/basic.xml; perhaps the rule name is mispelled or the rule doesn't exist anymore?"));
     }
 
+    /**
+     * When a custom ruleset references a ruleset that only contains deprecated rules, then this ruleset itself is
+     * considered deprecated and the user should get a deprecation warning for the ruleset.
+     */
+    @Test
+    public void testRuleSetReferencesDeprecatedRuleset() throws Exception {
```

Comment:
I think test methods get more reddable (scannable) when not every method is prefixed with test.